### ISSUE TITLE
savemat() would be a nice addition to this package

### ISSUE THREAD

**Issue by codeDogMcGee** [skjerns/mat7.3#21]
This is a very valuable package, as it reads my multi-GB Matlab 7.3 files and converts all of my data types quickly. In my case I need to append the data and save back to .mat 7.3 format as well, and have resorted to using the hdf5storage library, which does not load the files as cleanly as mat7.3. I would like to request adding a savemat() function, as the author says that would be easy to implement. Thanks for creating this awesome tool!

**Comment by skjerns** [comment]
I was a bit optistic when proposing that it would be easy to implement.

Which data types do you generally need? The problem is a bit, that datatypes don't map perfectly between the two languages. arrays are rather straight forward to implement, for the other datatypes such as cell/struct I need to dig a bit deeper on how exactly they are read by MATLAB.

However, if you just want to append data to a file, just using plain hdf5 might be a good choice, as it prevents you from reading the entire file (main purpose of hdf5)

**Comment by codeDogMcGee** [comment]
Thanks for the suggestion, I may just use hdf5 since I am only appending for now. My dataset does contain struct and cell data types.

**Comment by skjerns** [comment]
Yes, I think for this purpose it is much faster to hard code something and use hdf5 to prevent reading the entire file. However, that might also be a bit tricky to figure out how MATLAB accesses the referenced objects inside the `#ref#` object

**Comment by skjerns** [comment]
> Hi [@skjerns](https://github.com/skjerns) is this feature being looked into? I am currently trying to use loadmat from mat73 and using scipy savemat feature but 7.3 version is not supported.
> 
> I would like to take a look into adding this feature to mat73 if possible. Thanks.

nope, I've abandond the feature at this moment - if you want to look into it, feel free to do so!
